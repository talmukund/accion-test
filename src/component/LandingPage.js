import React, { Component } from "react";
import PropTypes from "prop-types";
import axios from "axios";
import logo from "./logo.svg";
import "./LandingPage.css";

export default class LandingPage extends Component {
  constructor(props){
    super(props);
    this.state = {
      userData : []
    }
    this.handleClick = this.handleClick.bind(this);
  }

  async componentDidMount(){
    const allPosts = await axios.get("https://jsonplaceholder.typicode.com/posts");
    this.setState({userData: allPosts.data});
  }

  handleClick (){
    this.props.history.push("/create")
  }

  render() {
    return (
      <div className="App">
        <header className="App-header">
          <img src={logo} className="App-logo" alt="logo" />
          <h1 className="App-title">Welcome to React</h1>
        </header>
        <button onClick={this.handleClick}>Move to create post</button>
        <p className="App-intro">
          To get started, edit <code>src/App.js</code> and save to reload.
        </p>
        <table>
          <tr>
            <th>userid</th>
            <th>id</th>
            <th>title</th>
            <th>body</th>
          </tr>
          
            {this.state.userData && this.state.userData.map(user=>{
              return<tr> <td>{user.userId}</td><td>{user.id}</td><td>{user.title}</td><td>{user.body}</td></tr>
            })}          
        </table>        
      </div>
    );
  }
}
