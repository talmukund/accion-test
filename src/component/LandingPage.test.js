import React from 'react';
import LandingPage from './LandingPage';
import toJson from 'enzyme-to-json'
import { shallow } from 'enzyme'

const props = {}

it('renders correctly', () => {
    const wrapper = shallow(<LandingPage />);
    const tree = toJson(wrapper);
    expect(tree).toMatchSnapshot();
});