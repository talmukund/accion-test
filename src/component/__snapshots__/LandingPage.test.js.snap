// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <img
      alt="logo"
      className="App-logo"
      src="logo.svg"
    />
    <h1
      className="App-title"
    >
      Welcome to React
    </h1>
  </header>
  <button
    onClick={[Function]}
  >
    Move to create post
  </button>
  <p
    className="App-intro"
  >
    To get started, edit 
    <code>
      src/App.js
    </code>
     and save to reload.
  </p>
  <table>
    <tr>
      <th>
        userid
      </th>
      <th>
        id
      </th>
      <th>
        title
      </th>
      <th>
        body
      </th>
    </tr>
  </table>
</div>
`;
